monkey/tests/unit_tests/infection_monkey/network_scanning/test_ping_scanner.py

```diff
@@ -86,56 +105,102 @@ def set_os_windows(monkeypatch):
     monkeypatch.setattr("sys.platform", "win32")
 
 
+@pytest.fixture
+def mock_agent_event_queue():
+    return MagicMock(spec=IAgentEventQueue)
+
+
+HOST_IP = "192.168.1.1"
+TIMEOUT = 1.0
+
+
+def _get_ping_scan_event(result: PingScanData):
+    return PingScanEvent(
+        source=get_agent_id(),
+        target=HOST_IP,
+        timestamp=TIMESTAMP,
+        tags=frozenset(),
+        response_received=result.response_received,
+        os=result.os,
+    )
+
+
 @pytest.mark.usefixtures("set_os_linux")
-def test_linux_ping_success(patch_subprocess_running_ping_with_ping_output):
+def test_linux_ping_success(patch_subprocess_running_ping_with_ping_output, mock_agent_event_queue):
     patch_subprocess_running_ping_with_ping_output(LINUX_SUCCESS_OUTPUT)
-    result = ping("192.168.1.1", 1.0)
+    result = ping(HOST_IP, TIMEOUT, mock_agent_event_queue)
+    event = _get_ping_scan_event(result)
```

Comment:
Don't rely on private methods. The `mock_agent_event_queue` is what you should use to capture the event.

We can do the following:

1. monkeypatch get_agent_id() 
2. monkeypatch time.time()
3. Create an event with the expected data
4. `assert_called_with()` expected event

Reply:
The private method is in the test not in the ping_scanner. It is returning the expected event.

Reply:
Ah, I see. My bad.